addons/test_mass_mailing/tests/test_mailing.py

```diff
@@ -378,3 +378,33 @@ def test_mailing_mailing_list_optout(self):
             check_mail=True
         )
         self.assertEqual(mailing.ignored, 2)
+
+    @users('user_marketing')
+    def test_mass_mailing_duplicates(self):
+        """
+        Tests whether function `_get_seen_list` is correctly able to identify duplicate emails,
+        even through different batches.
+        Mails use different names to make sure they are recognized as duplicates even without being
+        normalized (e.g.: '"jc" <0@example.com>' and '"vd" <0@example.com>' are duplciates)
+        """
+        BATCH_SIZE = 5
+        names = ['jc', 'vd']
+        emails = [f'{i}@example.com' for i in range(BATCH_SIZE)]
+        records = self.env['mailing.test.partner'].create([{
+            'name': f'test_duplicates {i}', 'email_from': f'"{names[i % 2]}" <{emails[i % BATCH_SIZE]}>'
+        } for i in range(20)])
+
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'test duplicates',
```

Comment:
:socks: Just to be annoying (as there are other suggestions), could you order keys alphabetically ? We try to use that order, to be coherent through tests as we need some sort of convention ^^